index.js

```diff
@@ -140,37 +196,96 @@ function loader (registryOrVersion) {
-      if (type === null) throw new Error("Don't know the serialized type for enchant level")
-      if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+      if (registry.type === 'bedrock') {
+        const enchs = normalizedEnchArray.map(({ name, lvl }) => ({
+          id: { type: 'short', value: registry.enchantmentsByName[name].id },
+          lvl: { type: 'short', value: lvl }
+        }))
 
-      const enchs = normalizedEnchArray.map(({ name, lvl }) => {
-        const value = type === 'short' ? registry.enchantmentsByName[name].id : `minecraft:${registry.enchantmentsByName[name].name}`
-        return { id: { type, value }, lvl: { type: 'short', value: lvl } }
-      })
+        if (enchs.length !== 0) {
+          if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+          this.nbt.value.ench = { type: 'list', value: { type: 'compound', value: enchs } }
+        }
+      } else if (registry.type === 'pc') {
+        const isBook = this.name === 'enchanted_book'
+        const enchListName = registry.supportFeature('nbtNameForEnchant')
+        const type = registry.supportFeature('typeOfValueForEnchantLevel')
+        if (type === null) throw new Error("Don't know the serialized type for enchant level")
+        if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
 
-      if (enchs.length !== 0) {
-        this.nbt.value[isBook ? 'StoredEnchantments' : enchListName] = { type: 'list', value: { type: 'compound', value: enchs } }
-      }
+        const enchs = normalizedEnchArray.map(({ name, lvl }) => {
+          const value =
+            type === 'short'
+              ? registry.enchantmentsByName[name].id
+              : `minecraft:${registry.enchantmentsByName[name].name}`
+          return { id: { type, value }, lvl: { type: 'short', value: lvl } }
+        })
+
+        if (enchs.length !== 0) {
+          this.nbt.value[isBook ? 'StoredEnchantments' : enchListName] = {
+            type: 'list',
+            value: { type: 'compound', value: enchs }
+          }
+        }
 
-      // The 'registry.itemsByName[this.name].maxDurability' checks to see if this item can lose durability
-      if (registry.supportFeature('whereDurabilityIsSerialized') === 'Damage' && registry.itemsByName[this.name].maxDurability) {
-        this.nbt.value.Damage = { type: 'int', value: 0 }
+        // The 'registry.itemsByName[this.name].maxDurability' checks to see if this item can lose durability
+        if (
+          registry.supportFeature('whereDurabilityIsSerialized') === 'Damage' &&
+          registry.itemsByName[this.name].maxDurability
+        ) {
+          this.nbt.value.Damage = { type: 'int', value: 0 }
+        }
       }
     }
 
+    // TODO: namespaces other than minecraft:
+    // The item palette on bedrock can contain items
+    // with other namespaces, probably need to handle
+    // this in prismarine-registry
+    // Also this needs to be tested
```

Comment:
We don't handle non-minecraft namespaces on Java Edition either, so there is no need to worry about it for bedrock. Our priority is vanilla for now, but if big problems arise it can be investigated later